### ISSUE TITLE
SmartThings information issues

### ISSUE THREAD

**Issue by Theiamania** [ollo69/ha-samsungtv-smart#8]
Hi Ollo,

I follow all steps (several times) but can't connect with my Samsung Q55 when I try to use the SmartThing token.
I can connect properly if keep blank the token field. But I need to use SmartThings Keys...

When I try to connect with the TV it ask meif I want grant access to HA, fine. But later HA shows the message "Wrong SmartThings token or TV not registered in SmartThings." and the integration is never done.
But the file token-192.168.0.10.txt are created and have an 8 digits number. I delete this file each time that I try to connect with a new token.

I have my iPhone connected to the TV, so ST works. No?
The TV is registered in Samsung ST. I know it because I can see the events list of my TV in Samsung web (Turned ON at this time, switched to HDMI2...).
I tried deleting and generating new tokens, don't works.
No log to show, home-assistant.log file don't have anything about this connection.

Any ideas?

**Comment by Aiace88** [comment]
idem, but the TV does not ask me for confirmation

**Comment by Theiamania** [comment]
Hi Aiace,

Try to remove HA from the device list in Config\General\External devices in your TV
Maybe it will ask you for confirmation.

**Comment by ollo69** [comment]
Sorry but is not clear how you generate ST token and seems that there is a little bit confunsion.
Token file generated in component dir have nothink to do with ST token. ST token must be generated connectig to ST web site and following the instruction provided in the main page. Did you follow this guide?

**Comment by Aiace88** [comment]
> Hi Aiace,
> 
> Try to remove HA from the device list in Config\General\External devices in your TV
> Maybe it will ask you for confirmation.

hi, are you referring to the television menu? there is only my phone between the devices

**Comment by Theiamania** [comment]
> Sorry but is not clear how you generate ST token and seems that there is a little bit confunsion.
> Token file generated in component dir have nothink to do with ST token. ST token must be generated connectig to ST web site and following the instruction provided in the main page. Did you follow this guide?

Hi, I follow step by step your guide in https://github.com/ollo69/ha-samsungtv-smart/blob/master/Smartthings.md and I have my token.
<img width="1390" alt="screenshot" src="https://user-images.githubusercontent.com/1210407/78773722-f2454200-7993-11ea-9f2f-9eada8ddf09b.png">

Because it didn't works for me, I tried to reconnect doing some changes like delete token file... Please, forget it.

When I reinstall your component, reboot HA, clear the device list in my TV, and follow all steps, I cant connect (but permission dialog is showed in the TV) and the message "Wrong SmartThings token or TV not registered in SmartThings" is showed.

I tried to make a manual connection editing the configuration.yaml but same result (dialog box in TV, no integration created) without any  log information.

**Comment by Theiamania** [comment]
> > Hi Aiace,
> > Try to remove HA from the device list in Config\General\External devices in your TV
> > Maybe it will ask you for confirmation.
> 
> hi, are you referring to the television menu? there is only my phone between the devices

Yes, on TV. This step works for me, even though the connection is not finished creating.

**Comment by SebastianPietschner** [comment]
I am also having issues with this. 
![tempsnip](https://user-images.githubusercontent.com/32213671/78867764-fdc46600-7a84-11ea-97e3-e0af55348dca.png)

This is what the device looks like under SmartThings. I had to remove and then add it again to have it connected to my account. Now there is less information.

**Comment by ollo69** [comment]
Problem is type. I'm filtering for "Samsung OCF TV". Or you change the type (with edit button) or I remove the filter but in this case I could associate a wrong device during setup,
Was this type associated automatically or you changed something manually?

![image](https://user-images.githubusercontent.com/60491700/78880715-18331b00-7a56-11ea-8be5-cadffd419dc5.png)

**Comment by SebastianPietschner** [comment]
Yes, I did use the app to associate it. How do you do it on the site?

**Comment by Theiamania** [comment]
Same as Sebastian. Associated by phone app and the Type result is "Placeholder".

I try to edit Type field to "Samsung OCF TV" and "Device Network Id" are required in this edition process. 

Searching about ST "Device Network Id" I found:
Device Network Id – this must be unique, but otherwise doesn’t matter. I just used the name of my device, in all caps and one word.

So I  try this:
<img width="471" alt="screenshot 2" src="https://user-images.githubusercontent.com/1210407/78885997-d1e1ba00-7a5d-11ea-90e3-93d47a7e0dce.png">

ha-samsungtv-smart can't connect. 
The message showed is: "Unable to connect to this Samsung TV device"

**Comment by ollo69** [comment]
I think that is not correctly registered. Remove it and register again using the phone app. I have a lot of information associated to my device that here is misssing.

**Comment by Theiamania** [comment]
This is my Device info after change to "Samsung OCF TV" Type
<img width="1698" alt="screenshot 3" src="https://user-images.githubusercontent.com/1210407/78887397-12dace00-7a60-11ea-893e-e9887a46dc92.png">

**Comment by ollo69** [comment]
This are correct information! In this way should work also with the control

**Comment by Theiamania** [comment]
Do you think that my "Device Network Id" is right? 
Or maybe i cant connect because "Device Network Id – this must be unique, but otherwise doesn’t matter. I just used the name of my device, in all caps and one word." is not correct?

When you setup your own TV, did you need to enter "Device Network Id"?

**Comment by ollo69** [comment]
No, I don't use this, but I don't now how SmartThings use this. I think that registration should be handled by the SmartThings apps and not manually by you. I suggest to remove this device and try to register agin, than use the portal only to check that is correctly registerd. I have a UUID in the field network id for my TV.

**Comment by Theiamania** [comment]
OK, I tried this:
1. Remove device list in TV
2. Remove TV in my ST account
3. Connect my TV to ST using my phone app
4. The new TV in my ST account have "placeholder" in Type field and no "Device Network Id" is showed.
<img width="1687" alt="screenshot" src="https://user-images.githubusercontent.com/1210407/78889874-d78ece00-7a64-11ea-8151-d99a047cab56.png">
5. Edit device manually. 
· Type -> "Samsung OCF TV" 
· Device Network Id -> "samsung_qled_55" 
<img width="467" alt="screenshot 2" src="https://user-images.githubusercontent.com/1210407/78890203-7adfe300-7a65-11ea-942e-d93a9cff0cb0.png">
6. Now Device Info are more detailed in the ST page
7. Delete and token and generate new one in Samsung account.
8. In HA, create new integration SamsungTV Smart
9. Grant access in TV side

Same result:  Wrong SmartThings token or TV not registered in SmartThings

**Comment by ollo69** [comment]
I do not understand why your TV is created with type "placeholder", I think that something is wrong in this step. May be is a problem on the app, or may be you are not using the correct procedure. TV should be automatically detected by the phone APP, is this occuring? Is your phone connected to the same wi-fi when you register the TV?

**Comment by Theiamania** [comment]
Yes, I just find a TV and the app find it. TV and phone are in the same WiFi.

**Comment by Theiamania** [comment]
Maybe this is a clue:

On TV, when select "[Home Assistant] QLED 55" in remote device list -> show details, I see available services for this device and only shows one: play media from mobile device.

Perhaps must have a "remote control" or something similar, no?

**Comment by Aiace88** [comment]
i have the same problem with my q70, if i register in home assistant without the api it's ok, but if i use the api i have the same problem. my tv is type placeholder.

**Comment by ollo69** [comment]
> Maybe this is a clue:
> 
> On TV, when select "[Home Assistant] QLED 55" in remote device list -> show details, I see available services for this device and only shows one: play media from mobile device.
> 
> Perhaps must have a "remote control" or something similar, no?

What you see in the TV is only related to local connection and have nothing to do with SmartThings. I'm quite sure that the problem is how TV is registered in SmartThings. My TV show all information and type is correct without the need of manual change on the web site.

**Comment by Krocko** [comment]
For me this integration was working. 
Since yesterday I don’t get the states from smartthings. I tried to re add this integration and get the same error.

**Comment by ollo69** [comment]
I think they are doing something on their server. I have the list of input with duplicated values, never see this before:

![image](https://user-images.githubusercontent.com/60491700/78916025-23ee0400-7a8d-11ea-8eca-1124f7db934d.png)

I just fixed the code in the component because with this broken list was creating wrong source list!!
Try to configuring manually on SmartThings the correct TV Type, but in "Device Network ID" put a generated uuid (you can generate [here](https://www.uuidgenerator.net/version4), maybe it work.
From my side SmartThings still working....

**Comment by Theiamania** [comment]
I have duplicated inputs too.

I generated uuid for Device Network Id and edited the configuration.yaml file for manual config. After reboot I have this:
Logger: custom_components.samsungtv_smart.config_flow
Source: custom_components/samsungtv_smart/config_flow.py:149 
First occurred: 6:37:13 PM (1 occurrences) 
Last logged: 6:37:13 PM

Error during setup of host 192.168.0.12 using configuration.yaml info. Reason: not_successful

**Comment by ollo69** [comment]
I do not understand what you do in configuration.yaml. Anyway it is better to do first configuration using web interface, then you can use configuration.yaml for additional value (like app and source list)

**Comment by ollo69** [comment]
![image](https://user-images.githubusercontent.com/60491700/78919643-81388400-7a92-11ea-8e17-c25827bccb80.png)

**Comment by Theiamania** [comment]
This is my configuration.yaml

samsungtv_smart:
  - host: 192.168.0.12
    name: Samsung QLED
    api_key: <my generated api key>

**Comment by Theiamania** [comment]
And this is my config in ST:
<img width="474" alt="screenshot" src="https://user-images.githubusercontent.com/1210407/78920095-2bb0a700-7a93-11ea-94ba-46d440c4b82d.png">

**Comment by Aiace88** [comment]
> And this is my config in ST:
> <img alt="screenshot" width="474" src="https://user-images.githubusercontent.com/1210407/78920095-2bb0a700-7a93-11ea-94ba-46d440c4b82d.png">

but the device network id in my situation is not displayed...

**Comment by ollo69** [comment]
Comment this and configure using web interface. After TV is configured you don't need to specify API_KEY or Name because are saved in the registry. You must only set the key (host) and the additional value (source_list, app_list, etc)

```
samsungtv_smart:
  - host: 192.168.0.12
    source_list: <your source list, if you want>
    app_list: <your app list, if you want>
    ...
```

**Comment by Aiace88** [comment]
i remove my tv from my profile smartthings and i reconfigure from app android, but i dont see the network id....

**Comment by ollo69** [comment]
Network ID and device type can be configured from web site: https://account.smartthings.com/login

**Comment by Theiamania** [comment]
> Comment this and configure using web interface. After TV is configured you don't need to specify API_KEY or Name because are saved in the registry. You must only set the key (host) and the additional value (source_list, app_list, etc)
> 
> ```
> samsungtv_smart:
>   - host: 192.168.0.12
>     source_list: <your source list, if you want>
>     app_list: <your app list, if you want>
>     ...
> ```

Sorry, I'm lost. Do you say to first include api_key, and later erase it?



> Network ID and device type can be configured from web site: https://account.smartthings.com/login

Really? Great! Where can I get it? I don't see where

**Comment by Aiace88** [comment]
> Network ID and device type can be configured from web site: https://account.smartthings.com/login

yes, i have modify the type in this site...but the network id is not present....

sorry, you are italian?

**Comment by ollo69** [comment]
> 
> Sorry, I'm lost. Do you say to first include api_key, and later erase it?

I'm saying to use Home Assistant web interface to configure the component and not configuration.yaml!! Settings -> Integration and + button.

Configuration.yaml will be used after to add additional info.

**Comment by Aiace88** [comment]
> > Network ID and device type can be configured from web site: https://account.smartthings.com/login
> 
> > yes, i have modify the type in this site...but the network id is not present....
> 
> > sorry, you are italian?
> 
> Yes, I'm Italian, but why you say that "network id" is not present, is in your piture!!

no, i quote the post of theiamania.
this is my configuration.
i test to insert the token in the network id, i see the information of tv, but the configuration is not ok
![2020-04-09 19_13_18-Edit Device](https://user-images.githubusercontent.com/35429685/78922068-34ef4300-7a96-11ea-80b6-3bee45bcee0b.jpg)

![2020-04-09 19_15_39-Impostazioni - Home Assistant](https://user-images.githubusercontent.com/35429685/78922260-87306400-7a96-11ea-9288-6bdd94a31c9d.jpg)

p.s. sono italiano anch'io.

**Comment by Theiamania** [comment]
> > Sorry, I'm lost. Do you say to first include api_key, and later erase it?
> 
> I'm saying to use Home Assistant web interface to configure the component and not configuration.yaml!! Settings -> Integration and + button.
> 
> Configuration.yaml will be used after to add additional info.

Yes, like in the readme. Always I use this method, but I tried manual way too.
Thanks for add this featured!

**Comment by ollo69** [comment]
Sorry, I prefer not use italian because this is a public post.
In the field "Device Network ID" you must put a generated UUID form here: https://www.uuidgenerator.net/ and not your SmartThings token. The SmartThings token you must use after, during component configuration.
I don't know if this work, but is the only suggestion that I can give. 
**I repeat that normally this values should be automatically generated during TV registration, so I think that something is not working on Samsung.**

**Comment by Theiamania** [comment]
I tried to get a Device Network Id in the https://account.smartthings.com/login but I have only this:

`metadata {
	definition (name: "QLED", namespace: "HA", author: "theiamania", cstHandler: true) {
		capability "Media Playback"
		capability "Media Playback Repeat"
		capability "Media Playback Shuffle"
		capability "Media Presets"
		capability "Media Track Control"
		capability "Samsung TV"
		capability "TV"
		capability "Tv Channel"
	}


	simulator {
		// TODO: define status and reply messages here
	}

	tiles {
		// TODO: define your main and details tiles here
	}
}

// parse events into attributes
def parse(String description) {
	log.debug "Parsing '${description}'"
	// TODO: handle 'playbackStatus' attribute
	// TODO: handle 'supportedPlaybackCommands' attribute
	// TODO: handle 'playbackRepeatMode' attribute
	// TODO: handle 'playbackShuffle' attribute
	// TODO: handle 'presets' attribute
	// TODO: handle 'supportedTrackControlCommands' attribute
	// TODO: handle 'volume' attribute
	// TODO: handle 'mute' attribute
	// TODO: handle 'pictureMode' attribute
	// TODO: handle 'soundMode' attribute
	// TODO: handle 'switch' attribute
	// TODO: handle 'messageButton' attribute
	// TODO: handle 'volume' attribute
	// TODO: handle 'channel' attribute
	// TODO: handle 'power' attribute
	// TODO: handle 'picture' attribute
	// TODO: handle 'sound' attribute
	// TODO: handle 'movieMode' attribute
	// TODO: handle 'tvChannel' attribute
	// TODO: handle 'tvChannelName' attribute

}

// handle commands
def setPlaybackStatus() {
	log.debug "Executing 'setPlaybackStatus'"
	// TODO: handle 'setPlaybackStatus' command
}

def play() {
	log.debug "Executing 'play'"
	// TODO: handle 'play' command
}

def pause() {
	log.debug "Executing 'pause'"
	// TODO: handle 'pause' command
}

def stop() {
	log.debug "Executing 'stop'"
	// TODO: handle 'stop' command
}

def fastForward() {
	log.debug "Executing 'fastForward'"
	// TODO: handle 'fastForward' command
}

def rewind() {
	log.debug "Executing 'rewind'"
	// TODO: handle 'rewind' command
}

def setPlaybackRepeatMode() {
	log.debug "Executing 'setPlaybackRepeatMode'"
	// TODO: handle 'setPlaybackRepeatMode' command
}

def setPlaybackShuffle() {
	log.debug "Executing 'setPlaybackShuffle'"
	// TODO: handle 'setPlaybackShuffle' command
}

def playPreset() {
	log.debug "Executing 'playPreset'"
	// TODO: handle 'playPreset' command
}

def nextTrack() {
	log.debug "Executing 'nextTrack'"
	// TODO: handle 'nextTrack' command
}

def previousTrack() {
	log.debug "Executing 'previousTrack'"
	// TODO: handle 'previousTrack' command
}

def volumeUp() {
	log.debug "Executing 'volumeUp'"
	// TODO: handle 'volumeUp' command
}

def volumeDown() {
	log.debug "Executing 'volumeDown'"
	// TODO: handle 'volumeDown' command
}

def setVolume() {
	log.debug "Executing 'setVolume'"
	// TODO: handle 'setVolume' command
}

def mute() {
	log.debug "Executing 'mute'"
	// TODO: handle 'mute' command
}

def unmute() {
	log.debug "Executing 'unmute'"
	// TODO: handle 'unmute' command
}

def setPictureMode() {
	log.debug "Executing 'setPictureMode'"
	// TODO: handle 'setPictureMode' command
}

def setSoundMode() {
	log.debug "Executing 'setSoundMode'"
	// TODO: handle 'setSoundMode' command
}

def on() {
	log.debug "Executing 'on'"
	// TODO: handle 'on' command
}

def off() {
	log.debug "Executing 'off'"
	// TODO: handle 'off' command
}

def showMessage() {
	log.debug "Executing 'showMessage'"
	// TODO: handle 'showMessage' command
}

def volumeUp() {
	log.debug "Executing 'volumeUp'"
	// TODO: handle 'volumeUp' command
}

def volumeDown() {
	log.debug "Executing 'volumeDown'"
	// TODO: handle 'volumeDown' command
}

def channelUp() {
	log.debug "Executing 'channelUp'"
	// TODO: handle 'channelUp' command
}

def channelDown() {
	log.debug "Executing 'channelDown'"
	// TODO: handle 'channelDown' command
}

def setTvChannel() {
	log.debug "Executing 'setTvChannel'"
	// TODO: handle 'setTvChannel' command
}

def channelUp() {
	log.debug "Executing 'channelUp'"
	// TODO: handle 'channelUp' command
}

def channelDown() {
	log.debug "Executing 'channelDown'"
	// TODO: handle 'channelDown' command
}

def setTvChannelName() {
	log.debug "Executing 'setTvChannelName'"
	// TODO: handle 'setTvChannelName' command
}`

I don't see any Device Network Id, but i have no idea how this develop works

**Comment by ollo69** [comment]
????

**Comment by Aiace88** [comment]
> Sorry, I prefer not use italian because this is a public post.
> In the field "Device Network ID" you must put a generated UUID form here: https://www.uuidgenerator.net/ and not your SmartThings token. The SmartThings token you must use after, during component configuration.
> I don't know if this work, but is the only suggestion that I can give.
> **I repeat that normally this values should be automatically generated during TV registration, so I think that something is not working on Samsung side.**

ok, i have generate the uuid from this site, i insert in the site samsung, and select type: Samsung OCF TV

now i have many information from the tv,  mac address, the channel, ecc....

i return in home assistant, insert the ip, name and token, but i have the same error.
![2020-04-09 19_36_49-Impostazioni - Home Assistant](https://user-images.githubusercontent.com/35429685/78924068-7df4c680-7a99-11ea-9747-d5c882c45744.jpg)

when i generate the token, you select the all options?

![2020-04-09 19_37_32-SmartThings  Add a little smartness to your things](https://user-images.githubusercontent.com/35429685/78924137-949b1d80-7a99-11ea-863e-7c005d209025.jpg)

thanks for the support!

**Comment by ollo69** [comment]
No idea. 
Just another tip: are sure that you set all correct permission to the generated token? Eventualy try to generate a new one a follow very well all the instruction
Later I will do some test reconfiguring my TV, but for me the problem is on samsung side. Network Id should be automatically generated and device type should assigned correctly during phone configuration.

**Comment by ollo69** [comment]
Token name you can put any...

![image](https://user-images.githubusercontent.com/60491700/78925948-88fd2600-7a9c-11ea-9752-7c0f614fbf2f.png)

**Comment by Aiace88** [comment]
> No idea.
> Just another tip: are sure that you set all correct permission to the generated token? Eventualy try to generate a new one and **follow very well all the instruction**!!
> Later I will do some test reconfiguring my TV, but for me the problem is on samsung side. Network Id should be automatically generated and device type should assigned correctly during phone configuration.

yes, i create a token here
![2020-04-09 19_57_43-SmartThings  Add a little smartness to your things](https://user-images.githubusercontent.com/35429685/78925999-a0d4aa00-7a9c-11ea-8273-1d9dd3e18db6.jpg)

and select all
![2020-04-09 19_58_16-SmartThings  Add a little smartness to your things](https://user-images.githubusercontent.com/35429685/78926019-a8944e80-7a9c-11ea-8546-c611394944a2.jpg)

**Comment by ollo69** [comment]
Finally I can confirm that there are issues on SmartThings side.
The reason for the error during configuration is related to a call to SmartThings API that is not returning anymore the list of devices. The point is that this is a [documented API](https://smartthings.developer.samsung.com/docs/api-ref/st-api.html#operation/getDevices), so this issue should not occur.
Also there are many states that are not updated anymore on the SmartThings server device.
I hope that this issue will be solved shortly, for now I put a warning at the beginning of the readme files.

Please on next days try again to configure your TV with ST and report here if it works again.

**Comment by Theiamania** [comment]
Thanks ollo, for your work, time and patience.

I will try to connect again in a few days and I will inform you when done.

**Comment by ollo69** [comment]
I just create a new release with a work-around for this issue.

Please give me your feed-back.

**Comment by Krocko** [comment]
I got the error `Wrong SmartThings token` with this new version. I also tried a new generated token.

**Comment by Krocko** [comment]
I generated the new device id with the tool. 
After update the device on the smartthings site with this id I have seen, what under "current states" the `di` and `pi` are different. But this id's are the right ones.

**Comment by ollo69** [comment]
Not really clear what you do, but fine to hear that is working...

**Comment by Krocko** [comment]
If you add a new device in the smartthings app, you got the entry "placeholder" as device. If you edit this device on the smartthings website and add the right tv model and the self generated uuid from the generating website, this network id is different from this too values. 
![image](https://user-images.githubusercontent.com/29394115/79017338-5c670e00-7b71-11ea-904d-0eda5ff48ee5.jpeg)

**Comment by ollo69** [comment]
Yes, but network id have nothing to do with configuration. I need Device ID that is the property "di".
Just another question: is your TV logged to SmartThings? Because my suspect is that is created as "placeholder" because you don't configure the connection on the TV!!!

**Comment by Aiace88** [comment]
It's work with the new instruction!

**Comment by Krocko** [comment]
> Just another question: is your TV logged to SmartThings? Because my suspect is that is created as "placeholder" because you don't configure the connection on the TV!!!

I think so the tv was showing the pin that you must enter in the smartthings app.

**Comment by ollo69** [comment]
You should check on the TV menu. There is an aption to login the TV.

**Comment by Theiamania** [comment]
It's work for me too. Thank you ollo!

**Comment by Krocko** [comment]
> You should check on the TV menu. There is an aption to login the TV.

Do you mean the Samsung account login?

**Comment by ollo69** [comment]
I should have found the cause of the original problem. With latest version deviceID should be automatically detected again. Issue was that response from SmartThings is changed!!!!

**Comment by Aiace88** [comment]
> I should have found the cause of the original problem. With latest version deviceID should be automatically detected again. Issue was that response from SmartThings is changed!!!!

ok, you think it's better to delete everything and redo it, or if it works with the manual insertion procedure leave it and not touch?

**Comment by ollo69** [comment]
Automatic dectection do the same thing that you did manually, the saved configuration in the registry at the end is the same. 
You can update the component and leave the configuration as is.

**Comment by Aiace88** [comment]
ok, i update all.

now in logs i have:

2020-04-11 18:05:09 ERROR (MainThread) [custom_components.samsungtv_smart.media_player] SamsungTV Smart - Error refreshing from SmartThings
2020-04-11 18:42:39 WARNING (MainThread) [custom_components.samsungtv_smart.media_player] Retrieving applications list from TV. It is suggested to define a manual app_list (see component documentation)!!!
2020-04-11 18:42:49 ERROR (MainThread) [custom_components.samsungtv_smart.media_player] Error retrieving application list from TV. Method will be retried.
2020-04-11 18:42:49 WARNING (MainThread) [homeassistant.helpers.entity] Updating state for media_player.q70 (<class 'custom_components.samsungtv_smart.media_player.SamsungTVDevice'>) took 10.068 seconds. Please report it to the custom component author.

**Comment by ollo69** [comment]
This logs I thing is related to network issue. This days network is overloaded. I didn't any change that can affect this error! I suppose they will dissapear when network works fine.
I also suppose that, error apart, the component is working, correct? For the update on the site, not always reflect the local changes...

**Comment by Aiace88** [comment]
hi, i update the component. 

when i restart ha. i have this error:

2020-04-16 11:01:52 WARNING (MainThread) [custom_components.samsungtv_smart.media_player] Retrieving applications list from TV. It is suggested to define a manual app_list (see component documentation)!!!
2020-04-16 11:01:55 ERROR (MainThread) [custom_components.samsungtv_smart.media_player] Error retrieving application list from TV. Method will be retried.

and the time to reboot ha is too high.

at this time, the tv is off.
is it better to configure the list of apps by hand?
![2020-04-16 11_16_51-● configuration yaml - Visual Studio Code](https://user-images.githubusercontent.com/35429685/79438603-090f1880-7fd4-11ea-9a19-71a453f9e25a.jpg)

**Comment by ollo69** [comment]
Yes, automatic generation of app list not always work and is time consuming. As you read in the warning you should define your app list with configuration.yaml. Read documentation for details.

**Comment by Aiace88** [comment]
ok, I'm doing, I don't understand from the documentation if I have to write all the various app and source lists below
![2020-04-16 11_50_15-● configuration yaml - Visual Studio Code](https://user-images.githubusercontent.com/35429685/79442190-8046ab80-7fd8-11ea-8309-dad126e0a3db.jpg)

**Comment by Aiace88** [comment]
i reboot ha with the configuration posted some minutes ago. 
and now the entity is not avalaible and have this in the logs:

![2020-04-16 12_14_40-home-assistant log - Visual Studio Code](https://user-images.githubusercontent.com/35429685/79444880-116b5180-7fdc-11ea-971a-f9d37bb9d9b9.jpg)

**Comment by ollo69** [comment]
You don't write your app list properly. Post it here so that I ca check.

**Comment by Aiace88** [comment]
> You don't write your app list properly. Post it here so that I ca check.

ok, I fixed, there were no quotes.
now i have no errors, and it seems to change apps perfectly.

an error: if I turn on the TV the mediaplayer signals the channel correctly. but if I go to an app and then press on digitaltv, it no longer reports it.

![2020-04-16 12_35_34-Panoramica - Home Assistant](https://user-images.githubusercontent.com/35429685/79446659-ddddf680-7fde-11ea-93c2-d1757d67272c.jpg)

**Comment by ollo69** [comment]
You should check what happen on smartthings web site for your device.
Sometimes do not update the status.

**Comment by Aiace88** [comment]
> You should check what happen on smartthings web site for your device.
> Sometimes do not update the status.

in the smarthings site it's ok...
![2020-04-16 12_51_14-Show Device](https://user-images.githubusercontent.com/35429685/79448015-09fa7700-7fe1-11ea-86f5-8ce1e33670b4.jpg)

maybe it's a limit of the custom media player?
try to set up source lists too?

**Comment by ollo69** [comment]
No, I don't think this is the problem.
But what happen on the device tvChannelName when you select an app?

**Comment by Aiace88** [comment]
> No, I don't think this is the problem.
> But what happen on the device tvChannelName when you select an app?

it's ok.
now in this situation, the change of channel not work, but if I turn the TV off and on again, the channels appear on the mediaplayer and the channel change works

**Comment by Aiace88** [comment]
update: in the site smarthings not update the channel...but in the app android yes.
remove the tv from smarthings?

**Comment by ollo69** [comment]
If you want to clean-up the situation, do the following in this exact order:
- remove TV from the APP
- remove TV from HA
- **logout TV from smartThings (from TV menu)**
- delete TV device from smartThings web site (if still there)
- **login TV to smartThings (from TV menu)**
- add TV to your phone app (and set your pref as location, name etc)
- generate a new SmartThings token
- add TV to HomeAssistant with the new token 
- check SmartThings site

**Comment by Aiace88** [comment]
> If you want to clean-up the situation, do the following in this exact order:
> 
> * remove TV from the phone APP
> * remove TV from HA
> * **logout TV from smartThings (from TV menu)**
> * delete TV device from smartThings web site (if still there)
> * **login TV to smartThings (from TV menu)**
> * add TV to your phone app (and set your pref as location, name etc)
> * generate a new SmartThings token
> * add TV to HomeAssistant with the new token
> * check SmartThings site

ok, now from the site smarthings it's ok.
the problem is only for the hdmi and tv.
i configure     source_list: '{"TV": "KEY_TV", "SHIELD": "KEY_HDMI"}'
but not working...and not logs...

p.s. can you have telegram?

**Comment by ollo69** [comment]
For source control you must use ST key, normal key do not work.
so use this: '{"TV": "ST_TV", "SHIELD": "ST_HDMI1"}'

Anyway I did some test, what I discover is that not always channel info in SmartThings is refreshed. To have the refresh of the channel working you have to open 1 time, after powering on the TV, the SmartThing apps and look for the current source. 
This happen from the last update that also broke the TV discovery, so it seems that SmartThings API is in development and something else can still change...
I searched throught documentation but I found nothing to solve this issue.

**Comment by skynet01** [comment]
> If you want to clean-up the situation, do the following in this exact order:
> 
>     * remove TV from the phone APP
> 
>     * remove TV from HA
> 
>     * **logout TV from smartThings (from TV menu)**
> 
>     * delete TV device from smartThings web site (if still there)
> 
>     * **login TV to smartThings (from TV menu)**
> 
>     * add TV to your phone app (and set your pref as location, name etc)
> 
>     * generate a new SmartThings token
> 
>     * add TV to HomeAssistant with the new token
> 
>     * check SmartThings site

I've been having exact same problem here with the new The Frame 2020 tv. I did those steps and tv still registers as "Placeholder" with blank network id. On top of it all it shows up as "inactive" in the web portal.

**Comment by Krocko** [comment]
@ollo69 Maybe you could ask there https://community.smartthings.com

**Comment by Aiace88** [comment]
> > If you want to clean-up the situation, do the following in this exact order:
> > ```
> > * remove TV from the phone APP
> > 
> > * remove TV from HA
> > 
> > * **logout TV from smartThings (from TV menu)**
> > 
> > * delete TV device from smartThings web site (if still there)
> > 
> > * **login TV to smartThings (from TV menu)**
> > 
> > * add TV to your phone app (and set your pref as location, name etc)
> > 
> > * generate a new SmartThings token
> > 
> > * add TV to HomeAssistant with the new token
> > 
> > * check SmartThings site
> > ```
> 
> I've been having exact same problem here with the new The Frame 2020 tv. I did those steps and tv still registers as "Placeholder" with blank network id.
> 
> * logged in on the tv, authenticated the app using pin etc.
> * I generate the network-id from your suggested link (i tried both v1 and v4 option).
> * Switching tv from placeholder to Samsung OCF TV populates all the fields and shows all details correctly

@ollo69 
yes, the same with my q70 2019.

but now, a day later, the problem has come back on the smarthings website, which always reports as a YouTube channel. while the tv is on a tv channel. 
Not update the status, yesterday.
I had solved it by removing it and putting it back in my smarthings profile ... but it is absurd to think of doing it continuously. 

![2020-04-17 09_23_22-Show Device](https://user-images.githubusercontent.com/35429685/79543093-2227d000-808d-11ea-886d-f0de4bf3506a.jpg)
someone have this problem?

**Comment by ollo69** [comment]
Keep in mind that this approach to SmartThings is not the official way to interact with device, have to be considered a work-aroud. In fact we are using the TV device  that create the SmartThings phone app, and from some time the device is not created anymore as TV but as "place-holder". Leaving aside the fact that Samsung itself has its problems managing TVs remotely (see Alexa skill), the correct approach would be to create an app which in turn creates a device with which to interact, but to do this HomeAssistant should be connected in Https and would greatly limit the use of the component. All I can do now is to eventually exclude the channel check via SmartThings with an option, everything else would require a more in-depth analysis of the whole SmartThings world. Any suggestions are still welcome...

**Comment by cpuks** [comment]
The only problem I have:
SamsungTV Smart - Error refreshing from SmartThings

Althrough component works fine - I can mute/unmute change vol and call different apps, and well twice / hour I've got that error - I think it has to be that way.

@ollo69 does your HA shows that in logs too?

**Comment by ollo69** [comment]
Yes, it happen, but this log is too aggressive and in many case unusefull. Just occur when network fail.
I have days when never occurs and days with many instance of the message,
I will manage in better way on next release

**Comment by Theiamania** [comment]
Same problems here.

> I will manage in better way on next release

So have you already found a better way to do it? Are you developing it already?

**Comment by Aiace88** [comment]
idem, i have many logs.

[custom_components.samsungtv_smart.media_player] SamsungTV Smart - Error refreshing from SmartThings

**Comment by cpuks** [comment]
Cool, glad to hear as I was chasing some problems today when HA became unresponsive on VM machine and looking into all logs now. Good work - cheers for that component!

**Comment by skynet01** [comment]
So is there a definitive set up guide or things are still broken on samsung's end? :)

**Comment by ollo69** [comment]
> So is there a definitive set up guide or things are still broken on samsung's end? :)

What do you mean? Did you istall the component? Is it working for you? 
The only current issue is that not always channel names / numbers are updated, but, as I know, the component is working. 
The only alternative option SmartThings side is to implement comunication as is implemented for HA SmartThings component, creating a dedicated SmartThings app. But this require that HA is published in the internet with HTTPS protocol and will limit a lot usage of the component, so I don't consider this a valid alternative.

**Comment by skynet01** [comment]
Looks like the issue was indeed with the Smarthings API. I tried it again today and everything worked right away from the first try. I checked the API and it looked like the tv entity was populated correctly on Samsung site this time as well. We can probably close this ticket as it was their fault... On another note ... for some reason, this particular new tv doesn't update state to Smartthings often. Hopefully, they will fix this at some point...
thanks again for your support!

**Comment by ollo69** [comment]
I don't close this issue, because I'm really not covinced by the use of SmartThings and I need some feedback to better understand
I deleted and recreated my TV on SmartThings apps and now TVChannel are not updating anymore.
Many user are reporting the same, so in last component version I disabled the use of channel info (can be enabled again using the "option" button in integration page).
I'm not understanding if this is really a SmartThing issue or what really Samsung want to do.
May be different TV model have different behavior.
Pls report here your feed-back to identify the best way to use SmartThings
